lighthouse-core/test/lib/tracehouse/task-summary-test.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint-env jest */
+
+import {
+  getJavaScriptURLs,
+  getAttributableURLForTask,
+  getExecutionTimingsByURL,
+} from '../../../lib/tracehouse/task-summary.js';
+import NetworkRecorder from '../../../lib/network-recorder.js';
+import MainThreadTasks from '../../../lib/tracehouse/main-thread-tasks.js';
+import ampTrace from '../../fixtures/traces/amp-m86.trace.json';
+import ampDevtoolsLog from '../../fixtures/traces/amp-m86.devtoolslog.json';
+import TraceProcessor from '../../../lib/tracehouse/trace-processor.js';
+import networkRecordsToDevtoolsLog from '../../network-records-to-devtools-log.js';
+import {taskGroups} from '../../../lib/tracehouse/task-groups.js';
+
+function getTasks(trace) {
+  const {mainThreadEvents, frames, timestamps} = TraceProcessor.processTrace(trace);
+  return MainThreadTasks.getMainThreadTasks(mainThreadEvents, frames, timestamps.traceEnd);
+}
+
+describe('Task Summaries', () => {
```

Comment:
these functions are quite well tested by the existing BootupTime et al tests, but only as integrated into the audit results, no unit tests, so it seemed appropriate to give some basic coverage since they're often used as units.